src/mono/wasm/runtime/loader/assets.ts

```diff
@@ -67,58 +89,70 @@ export function shouldLoadIcuAsset(asset: AssetEntryInternal): boolean {
     return !(asset.behavior == "icu" && asset.name != loaderHelpers.preferredIcuAsset);
 }
 
-function getSingleAssetWithResolvedUrl(resources: ResourceList | undefined, behavior: SingleAssetBehaviors): AssetEntry {
-    const keys = Object.keys(resources || {});
+function convertSingleAsset(modulesAssets: AssetEntryInternal[], resource: ResourceList | undefined, behavior: SingleAssetBehaviors): AssetEntryInternal {
```

Comment:
Is it right that 'resources: ResourceList' (plural, a list) is now 'resource' (singular, still a list?)

Reply:
The `ResourceList` name is not great. It's actually json dictionary, string key, string value. The key is file name and value is optional hash. And it's used as list of files, so the name makes some sense. 

Some asset types/behaviors have just one possible file. So this is a special case of `ResourceList` which has just one key/file in it.